src/codeu/chat/server/BroadCastSystem.java

```diff
@@ -0,0 +1,148 @@
+package codeu.chat.server;
+
+import codeu.chat.common.*;
+import codeu.chat.util.Serializers;
+import codeu.chat.util.connections.Connection;
+import codeu.chat.util.Uuid;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * Created by rsharif on 3/17/17.
+ */
+
+
+public class BroadCastSystem {
+
+    private class ConversationMessageLink {
```

Comment:
I could store the references to the conversation object but I'm not sure I understand what you mean in terms of the benefit. The reason I set it up this way is because the HashMap that keeps track of which users are in which conversations maps the Uuid to the Connections. As such, I never use the Conversation object..just its Uuid to find which connections are interested in these messages.